crates/red_knot_python_semantic/src/symbol.rs

```diff
@@ -593,8 +593,18 @@ fn symbol_by_id<'db>(
                 "__slots__" | "TYPE_CHECKING"
             );
 
-            widen_type_for_undeclared_public_symbol(db, inferred, is_considered_non_modifiable)
-                .into()
+            if scope.is_module_scope(db) && scope.file(db).is_stub(db.upcast()) {
+                // We generally trust module-level undeclared symbols in stubs and do not
+                // union with `Unknown`. If we don't do this, simple type aliases like
```

Comment:
nit: _type_ aliases are pretty consistently annotated with `: TypeAlias` in typeshed. But by annotating a symbol with `TypeAlias`, you're signalling to the type checker that the right-hand side of the assignment should be interpreted as a type expression and that the symbol should only be usable in contexts where type expressions are accepted. If the `IOError = OSError` assignment in typeshed were chnaged to `IOError: TypeAlias = OSError`, for example, type checkers might start disallowing uses of `IOError` in contexts where value expressions are expected, such as `X = IOError(42)`; that wouldn't be correct.

So arguably the issue being fixed here doesn't concern type aliases at all: it concerns all aliases that are _not_ type aliases!